File: HIL_testing/protoip_projects/heterogeneous_3/ip_design/src/foo_user.cpp
/* 
* icl::protoip
* Author: asuardi <https://github.com/asuardi>
* Date: November - 2014
*/

#include "foo_data.h"
#include "user_protoip_definer.h"
#include "user_main_header.h"
#include "user_nnz_header.h"
#include "user_structure_header.h"
#include "user_prototypes_header.h"



void foo_user(  data_t_minres_data_in minres_data_in_int[MINRES_DATA_IN_LENGTH],
		 	 	part_matrix *block_in_int,
				d_type_lanczos out_block_in_int[OUT_BLOCK_IN_LENGTH],
				data_t_x_in_in x_in_in_int[X_IN_IN_LENGTH],
				data_t_y_out_out y_out_out_int[Y_OUT_OUT_LENGTH])
{

	minres_HW(block_in_int, out_block_in_int, x_in_in_int, y_out_out_int, minres_data_in_int);

}

File: HIL_testing/protoip_projects/heterogeneous_3/ip_design/src/foo_user.m
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% icl::protoip
% Author: asuardi <https://github.com/asuardi>
% Date: November - 2014
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


function [y_out_out_int] = foo_user(project_name,minres_data_in_int, block_in_int, out_block_in_int, x_in_in_int)


	% load project configuration parameters: input and output vectors (name, size, type, NUM_TEST, TYPE_TEST)
	load_configuration_parameters(project_name);

	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
	% compute with Matlab and save in a file simulation results y_out_out_int
	for i=1:Y_OUT_OUT_LENGTH
		y_out_out_int(i)=0;
		for i_minres_data = 1:MINRES_DATA_IN_LENGTH
			y_out_out_int(i)=y_out_out_int(i) + minres_data_in_int(i_minres_data);
		end
		for i_block = 1:BLOCK_IN_LENGTH
			y_out_out_int(i)=y_out_out_int(i) + block_in_int(i_block);
		end
		for i_out_block = 1:OUT_BLOCK_IN_LENGTH
			y_out_out_int(i)=y_out_out_int(i) + out_block_in_int(i_out_block);
		end
		for i_x_in = 1:X_IN_IN_LENGTH
			y_out_out_int(i)=y_out_out_int(i) + x_in_in_int(i_x_in);
		end
	end

end

